src/jdk.zipfs/share/classes/jdk/nio/zipfs/ZipFileSystem.java

```diff
@@ -1383,33 +1439,33 @@ private void removeFromTree(IndexNode inode) {
      * Checks if the Zip File System property "releaseVersion" has been specified. If it has,
      * use its value to determine the requested version. If not use the value of the "multi-release" property.
      */
-    private void initializeReleaseVersion(Map<String, ?> env) throws IOException {
+    private Optional<Integer> determineReleaseVersion(Map<String, ?> env) throws IOException {
         Object o = env.containsKey(PROPERTY_RELEASE_VERSION) ?
             env.get(PROPERTY_RELEASE_VERSION) :
             env.get(PROPERTY_MULTI_RELEASE);
 
-        if (o != null && isMultiReleaseJar()) {
-            int version;
-            if (o instanceof String) {
-                String s = (String)o;
-                if (s.equals("runtime")) {
-                    version = Runtime.version().feature();
-                } else if (s.matches("^[1-9][0-9]*$")) {
-                    version = Version.parse(s).feature();
-                } else {
-                    throw new IllegalArgumentException("Invalid runtime version");
-                }
-            } else if (o instanceof Integer) {
-                version = Version.parse(((Integer)o).toString()).feature();
-            } else if (o instanceof Version) {
-                version = ((Version)o).feature();
+        if (o == null || !isMultiReleaseJar()) {
+            return Optional.empty();
+        }
+        int version;
+        if (o instanceof String) {
```

Comment:
Any reason to not use:

> 
    if (o instanceof String s) {
        if (s.equals("runtime")) {
            version = Runtime.version().feature();
        } else else if (s.matches("^[1-9][0-9]*$")) { {
            version = Version.parse(s).feature();;
            }
    }